app/routes/characters.tsx

```diff
@@ -57,34 +59,84 @@ export const meta: MetaFunction = () => {
 }
 
 type LineItemProps = {
-  label: string,
-  value: string
-}
+  label: string;
+  value: string;
+};
 
 function LineItem ({ label, value }: LineItemProps) {
   return (
-    <p className="flex justify-between items-baseline border-b-2 mb-2 pb-2 text-lg font-semibold">{label} <span className="font-normal text-base">{value}</span></p>
+    <p className="flex justify-between items-baseline border-b-2 mb-2 pb-2 text-lg font-semibold">
+      {label} <span className="font-normal text-base">{value}</span>
+    </p>
   )
 }
 
 let pageNumber = 1
 
 export default function CharactersRoute (): ReactNode {
   const data = useLoaderData<Character[]>()
-  const [results, setResults] = React.useState(data)
+
+  const [results, setResults] = useState(data)
+  const [scrollingDown, setScrollingDown] = useState(true)
+
   const fetcher = useFetcher()
 
-  React.useEffect(() => {
+  useEffect(() => {
     if (fetcher.data) {
-      setResults((prevResults: Character[]) => [...prevResults, ...fetcher.data])
+      setResults((prevResults: Character[]) => [
+        ...prevResults,
+        ...fetcher.data
+      ])
     }
   }, [fetcher.data])
+
+  useEffect(() => {
+    let scrollPosition = 0
+    let scrollDirection
+
+    window.addEventListener('scroll', function (e) {
+      scrollDirection = (document.body.getBoundingClientRect()).top > scrollPosition ? 'up' : 'down'
+      scrollPosition = (document.body.getBoundingClientRect()).top
+      setScrollingDown(scrollDirection === 'down')
+    })
+    return () => {
+      window.removeEventListener('scroll', function (e) {
+        scrollDirection = (document.body.getBoundingClientRect()).top > scrollPosition ? 'up' : 'down'
+        scrollPosition = (document.body.getBoundingClientRect()).top
+        setScrollingDown(scrollDirection === 'down')
+      })
+    }
+  }, [])
+
+  const scrollTo = (x: number, y: number) => {
+    window.scrollTo(x, y)
+  }
+  const scrollToTop = () => {
+    scrollTo(0, 0)
```

Comment:
```suggestion
    window.scrollTo(0, 0)
```

nit: we can inline here since it's only a passthrough, no need for the extra helper :)